internal/state/store.go

```diff
@@ -524,39 +523,43 @@ func (store dbStore) LoadFinalizeBlockResponse(height int64) (*abci.FinalizeBloc
 // This method is used for recovering in the case that we called the Commit ABCI
 // method on the application but crashed before persisting the results.
 func (store dbStore) LoadLastFinalizeBlockResponse(height int64) (*abci.FinalizeBlockResponse, error) {
```

Comment:
Not sure I understand the crash scenario fully. And this is the first time I am looking at this code. With the new change if `height` is smaller than the height of the `lastABCIResponseKey` it will return stored results (or not found if  `DiscardABCIResponses` was true). Maybe this is not possible in the crash scenario. If that's the case, could you maybe document this?

Reply:
> With the new change if height is smaller than the height of the lastABCIResponseKey

This is not possible IMHO. I.e. if we know we crashed at height 5, then we'll try to restore last responses for this height.